buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,404 @@
+@Library('dotnet-ci') _
+
+// Incoming parameters.  Access with "params.<param name>".
+// Note that the parameters will be set as env variables so we cannot use names that conflict
+// with the engineering system parameter names.
+
+//--------------------- Windows Functions ----------------------------//
+
+def windowsBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoBuildFlag = ((pgo == 'nopgo') ? '-nopgooptimize' : '-enforcepgo')
+    String baselineString = ""
+
+    // For baseline builds, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        bat "git checkout HEAD^^1"
+    }
+
+    bat "set __TestIntermediateDir=int&&.\\build.cmd -${config} -${arch} -skipbuildpackages ${pgoBuildFlag}"
+
+    // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
+    stash name: "nt-${arch}-${pgo}${baselineString}-build-artifacts", includes: 'bin/**'
+    stash name: "nt-${arch}-${pgo}${baselineString}-test-artifacts", includes: 'bin/tests/**'
+}
+
+def windowsPerf(String arch, String config, String uploadString, String runType, String opt_level, String jit, String pgo, String scenario, boolean isBaseline) {
+    withCredentials([string(credentialsId: 'CoreCLR Perf BenchView Sas', variable: 'BV_UPLOAD_SAS_TOKEN')])
+    
+    checkout scm
+    String baselineString = ""
+    if (isBaseline) {
+        baselineString = "-baseline"
+    }
+    dir ('.') {
+        unstash "nt-${arch}-${pgo}${baselineString}-build-artifacts"
+        unstash "benchview-tools"
+        unstash "metadata"
+    }
+
+    String pgoTestFlag = ((pgo == 'nopgo') ? '-nopgo' : '')
+
+    // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+    // submission-metadata.py
+    if (isBaseline) {
+        bat "move /y submission-metadata-baseline.json submission-metadata.json"
+    }
+
+    String testEnv = ""
+
+    if (jit == 'legacy_backend') {
+        testEnv = "-testEnv %WORKSPACE%\\tests\\legacyjit_x86_testenv.cmd"
+    }
+
+    String failedOutputLogFilename = "run-xunit-perf-scenario.log"
+
+    bat "py \".\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\""
+    bat ".\\init-tools.cmd"
+    bat "run.cmd build -Project=\"tests\\build.proj\" -BuildOS=Windows_NT -BuildType=${config} -BuildArch=${arch} -BatchRestorePackages"
+    bat "tests\\runtest.cmd ${config} ${arch} GenerateLayoutOnly"
+
+    // We run run-xunit-perf differently for each of the different job types
+    if (scenario == 'perf') {
+        String runXUnitPerfCommonArgs = "-arch ${arch} -configuration ${config} -generateBenchviewData \"%WORKSPACE%\\Microsoft.Benchview.JSONFormat\\tools\" ${uploadString} -runtype ${runType} ${testEnv} -optLevel ${opt_level} ${pgoTestFlag} -jitName ${jit} -stabilityPrefix \"START \"CORECLR_PERF_RUN\" /B /WAIT /HIGH /AFFINITY 0x2\""
```

Comment:
Is the plan to add the -uploadToBenchview flag here when we are ready to merge?

Reply:
The uploadString has already been updated.

Reply:
Hmm, must have been  looking at an old diff.